service/ibucketfactory.go

```diff
@@ -0,0 +1,35 @@
+package service
+
+import (
+	"context"
+	"io"
+	"mime/multipart"
+)
+
+//IBucketFactory inteface for the factory
+type IBucketFactory interface {
+	Get(objectName string, bucketName string) (io.Reader, error)
+	Put(uploadedFile *multipart.FileHeader, bucketName string, file multipart.File) (int64, error)
+}
+
+//BucketType declaration
+type BucketType int
+
+//Bucket Type enums
+const (
+	Google = iota
```

Comment:
Is there a particular reason why google has index `iota` ?

Reply:
this is a way to create an enum in golang. See here https://yourbasic.org/golang/iota/